src/json.ts

```diff
@@ -25,7 +26,10 @@ import {
 
 export const JsonStruct = define<Json>('Json', (value) => {
   const [isValid] = validateJsonAndGetSize(value, true);
-  return isValid;
+  if (!isValid) {
+    return 'Expected valid JSON value';
```

Comment:
Maybe to clarify this a bit further, "Expected a valid JSON-serializable value"?